tests/test_io.py

```diff
@@ -138,19 +138,28 @@ def test_reverse_readline(self):
             for idx, line in enumerate(reverse_readline(f)):
                 assert line == f"{str(self.NUMLINES - idx)}{os.linesep}"
 
-    def test_fake_big_file(self):
+    @pytest.mark.parametrize("l_end", ["\n", "\r\n"])
+    def test_big_file(self, l_end):
```

Comment:
This failure is a huge issue (reading big file is the purpose of this function), perhaps how the buffer is refilled or the pointer is advanced has two minor issue (don't know exactly ATM, haven't looked closely yet), leading to off-by-one error in generated line.

Really need to test this thoroughly, and test pymatgen against this before releasing.